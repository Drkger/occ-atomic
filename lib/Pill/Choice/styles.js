import colors from '../../subatomic/colors';
import fonts from '../../subatomic/fonts';
import spacing from '../../subatomic/spacing';

export default {
    pill: {
        background:colors.bgWhite,
        border:`1px solid ${colors.grey200}`,
        color:colors.ink,
        fontFamily:fonts.body,
        fontSize:14,
        fontWeight:400,
        lineHeight:'24px',
        height:32,
        padding:`0 ${spacing.base}px`,
        marginBottom:spacing.tiny,
        position:'relative',
        zIndex:0,
        outline:0,
        transition:'0.3s all',
        cursor:'pointer',
        borderRadius:spacing.small,
        '&:not(:last-child)': {
            marginRight:spacing.tiny
        },
        '&:hover': {
            background:colors.grey50
        }
    },
    selected: {
        background:`${colors.primLighter} !important`,
        zIndex:1,
        borderColor:colors.prim
    },
    disabled: {
        background:colors.white,
        borderColor:colors.grey100,
        color:colors.grey200,
        pointerEvents:'none'
    }
};
