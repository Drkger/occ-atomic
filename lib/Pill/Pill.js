import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Icon from '../Icon';

/** Group of buttons to toggle. */
class Pill extends Component {
    
    constructor(props) {
        super(props);
        this.state = {
            status: 'default',
            value: props.valueProp ? props.valueProp : undefined,
            touched: false
        };
        this.onClick = this.onClick.bind(this);
    }
    
    componentWillReceiveProps(props) {
        let { input, valueProp } = props;
        if (input && input.value)
            this.setState({ value: input.value });
        else if (valueProp != this.state.value)
            this.setState({ value: valueProp });
    }
    
    onClick(value) {
        const { onChange } = this.props;
        this.setState({
            value,
            touched: true
        });
        if (onChange) onChange(value);
    }
    
    render() {
        const { value, status, touched } = this.state;
        const { classes, label, assistiveText, group, meta, error, allowError, disabled, lockHeight } = this.props;
        let realStatus = status;
        const errorStatus = (meta && meta.error && (meta.touched || allowError)) || (error && (touched || allowError));
        if (disabled)
            realStatus = 'disabled';
        else if (errorStatus)
            realStatus = 'error';
        return (
            <div className={`${classes.container}${realStatus == 'disabled' ? ` ${classes.disabled}` : ''}`}>
                {(label || lockHeight) && (
                    <div className={classes.top}>
                        {label && (
                            <label className={`${classes.label} ${classes.left}`}>
                                {label}
                            </label>
                        )}
                    </div>
                )}
                <div className={classes.pillGroup}>
                    {group.map((pill, index) => (
                        <button key={pill.id} className={`${classes.pill}${value == pill.value ? ` ${classes.selected}` : ''}`} onClick={value != pill.value ? () => {this.onClick(pill.value, index);} : null}>{pill.label}</button>
                    ))}
                </div>
                {(assistiveText || lockHeight) && (
                    <div className={classes.bottom}>
                        {assistiveText && (
                            <label className={`${classes.label} ${classes.left} ${realStatus == 'error' ? classes.errorAssistiveText : classes.assistiveText}`}>
                                {realStatus == 'error' ? <Icon iconName="warning" width={14} height={14} className={classes.errorIcon} /> : null} {assistiveText}
                            </label>
                        )}
                    </div>
                )}
            </div>
        );
    }
}

Pill.propTypes = {
    classes: PropTypes.object.isRequired,
    /** Label text to show above the button's group. */
    label: PropTypes.string,
    /** Assistive text to show below the button's group. This text turns into an error message when the component receives the error prop. */
    assistiveText: PropTypes.string,
    /** Array of items to render the buttons. Each item must have an id (unique), a value (unique) and a label. */
    group: PropTypes.array,
    /** Value received from the parent component to override the child's component value. */
    valueProp: PropTypes.string,
    /** Object sended by react-form. */
    input: PropTypes.object,
    /** Metadata sended by react-form. */
    meta: PropTypes.object,
    /** Function to call when selecting an option. */
    onChange: PropTypes.func,
    /** After doing your validations outside of this component, you can change the state of the Pill to 'error' with this bool property. */
    error: PropTypes.bool,
    /** It allows the component to display the error messages even if it hasn't been touched. */
    allowError: PropTypes.bool,
    /** Disable the pill group. */
    disabled: PropTypes.bool,
    /** By default the Pill container would be smaller without a label or an assistive text. lockHeight will maintain the same height as if it had both, label and assistive text. */
    lockHeight: PropTypes.bool
};

export default Pill;