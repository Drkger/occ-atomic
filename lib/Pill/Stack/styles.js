import colors from '../../subatomic/colors';
import fonts from '../../subatomic/fonts';
import spacing from '../../subatomic/spacing';
import icons from '../../subatomic/icons';

export default {
    pill: {
        background:colors.bgWhite,
        border:`1px solid ${colors.grey200}`,
        fontFamily:fonts.body,
        fontSize:14,
        lineHeight:'24px',
        height:32,
        padding:`0 ${spacing.gutter}px`,
        marginBottom:spacing.tiny,
        position:'relative',
        zIndex:0,
        outline:0,
        flex:1,
        transition:'0.3s all',
        cursor:'pointer',
        borderRadius:spacing.small,
        '&:not(:last-child)': {
            marginRight:spacing.tiny
        },
        '&:not(:first-child)': {
            marginLeft:-1
        },
        '&:hover': {
            background:colors.grey50
        }
    },
    disabled: {
        background:colors.white,
        borderColor:colors.grey100,
        color:colors.grey200,
        pointerEvents:'none'
    },
    closeIcon: {
        width:spacing.small,
        height:spacing.small,
        display:'inline-block',
        background:icons.base(icons.close.icon([colors.grey300])),
        marginBottom:-4,
        marginLeft:2
    }
};
