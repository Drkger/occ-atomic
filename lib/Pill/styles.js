import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import HexToRgba from 'hex-rgba';

export default {
    container: {
        position:'relative',
        marginBottom:8
    },
    top: {
        height:24,
        '&:after': {
            content:'""',
            clear:'both'
        }
    },
    label: {
        fontFamily:fonts.body,
        fontSize:12,
        lineHeight:'20px',
        color:colors.ink,
        transition:'0.3s all',
        '& a': {
            color:colors.blue,
            textDecoration:'none',
            outline:'none'
        }
    },
    left: {
        float:'left'
    },
    pillGroup: {
        display:'flex'
    },
    disabled: {
        '& $label': {
            color:colors.inkLightest
        },
        '& $pill': {
            background:colors.white,
            borderColor:colors.grey100,
            color:colors.grey200,
            pointerEvents:'none'
        }
    },
    pill: {
        background:colors.bgWhite,
        border:`1px solid ${colors.grey200}`,
        fontFamily:fonts.body,
        fontSize:14,
        lineHeight:'24px',
        height:32,
        padding:0,
        position:'relative',
        zIndex:0,
        outline:0,
        flex:1,
        transition:'0.3s all',
        cursor:'pointer',
        '&:first-child': {
            borderRadius:'16px 0 0 16px'
        },
        '&:last-child': {
            borderRadius:'0 16px 16px 0'
        },
        '&:not(:first-child)': {
            marginLeft:-1
        },
        '&:hover': {
            background:colors.grey50
        }
    },
    selected: {
        background:[HexToRgba(colors.prim, 10), '!important'],
        zIndex:1,
        borderColor:colors.prim
    },
    bottom: {
        height:20,
        marginTop:4
    },
    assistiveText: {
        color:colors.inkLighter
    },
    errorAssistiveText: {
        color:colors.errorText
    },
    errorIcon: {
        marginBottom:-2
    }
};