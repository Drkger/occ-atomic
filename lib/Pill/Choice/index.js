import injectSheet from 'react-jss';
import Choice from './Choice';
import styles from './styles';

export default injectSheet(styles)(Choice);
