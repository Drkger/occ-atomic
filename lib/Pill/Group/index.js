import injectSheet from 'react-jss';
import Group from './Group';
import styles from './styles';

export default injectSheet(styles)(Group);
