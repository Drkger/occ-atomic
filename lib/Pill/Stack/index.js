import injectSheet from 'react-jss';
import Stack from './Stack';
import styles from './styles';

export default injectSheet(styles)(Stack);
