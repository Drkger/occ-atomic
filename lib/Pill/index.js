import injectSheet from 'react-jss';
import Pill from './Pill';
import styles from './styles';

export default injectSheet(styles)(Pill);