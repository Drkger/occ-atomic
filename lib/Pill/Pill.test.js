import React from 'react';
import { shallow } from 'enzyme';
import Pill from './Pill';
import styles from './styles';
import PillJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

const pills = [
    {
        id:1,
        value:10,
        label:'foo'
    },
    {
        id:2,
        value:20,
        label:'bar'
    },
    {
        id:3,
        value:30,
        label:'baz'
    },
    {
        id:4,
        value:40,
        label:'qux'
    },
];

describe("Pill", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <Pill group={pills} classes={classes} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('shows four buttons', () => {
        const wrapper = shallow(
            <Pill group={pills} classes={classes} />
        );
        expect(wrapper.find('.pill').length).toBe(4);
    });
    
    it('selects the right options', () => {
        const wrapper = shallow(
            <Pill group={pills} classes={classes} />
        );
        expect(wrapper.state('value')).toBe(undefined);
        wrapper.find('.pill').at(0).simulate('click');
        expect(wrapper.state('value')).toBe(10);
    });
    
    it('calls the onChange function', () => {
        const onChange = jest.fn();
        const wrapper = shallow(
            <Pill group={pills} onChange={onChange} classes={classes} />
        );
        wrapper.find('.pill').at(0).simulate('click');
        expect(onChange.mock.calls.length).toBe(1);
    });

});

describe("PillJSS", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <PillJSS group={pills} />
        );
        expect(wrapper).toMatchSnapshot();
    });

});

describe("Pill styles", () => {

    it('matches the snapshot', () => {
        expect(styles).toMatchSnapshot();
    });

});