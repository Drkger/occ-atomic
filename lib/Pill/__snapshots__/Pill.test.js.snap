// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pill
    classes={
      Object {
        "assistiveText": "assistiveText",
        "bottom": "bottom",
        "container": "container",
        "disabled": "disabled",
        "errorAssistiveText": "errorAssistiveText",
        "errorIcon": "errorIcon",
        "label": "label",
        "left": "left",
        "pill": "pill",
        "pillGroup": "pillGroup",
        "selected": "selected",
        "top": "top",
      }
    }
    group={
      Array [
        Object {
          "id": 1,
          "label": "foo",
          "value": 10,
        },
        Object {
          "id": 2,
          "label": "bar",
          "value": 20,
        },
        Object {
          "id": 3,
          "label": "baz",
          "value": 30,
        },
        Object {
          "id": 4,
          "label": "qux",
          "value": 40,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <div
          className="pillGroup"
        >
          <button
            className="pill"
            onClick={[Function]}
          >
            foo
          </button>
          <button
            className="pill"
            onClick={[Function]}
          >
            bar
          </button>
          <button
            className="pill"
            onClick={[Function]}
          >
            baz
          </button>
          <button
            className="pill"
            onClick={[Function]}
          >
            qux
          </button>
        </div>,
        undefined,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="pill"
              onClick={[Function]}
            >
              foo
            </button>,
            <button
              className="pill"
              onClick={[Function]}
            >
              bar
            </button>,
            <button
              className="pill"
              onClick={[Function]}
            >
              baz
            </button>,
            <button
              className="pill"
              onClick={[Function]}
            >
              qux
            </button>,
          ],
          "className": "pillGroup",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "foo",
              "className": "pill",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "foo",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": "bar",
              "className": "pill",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "bar",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": "baz",
              "className": "pill",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "baz",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "host",
            "props": Object {
              "children": "qux",
              "className": "pill",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "qux",
            "type": "button",
          },
        ],
        "type": "div",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <div
            className="pillGroup"
          >
            <button
              className="pill"
              onClick={[Function]}
            >
              foo
            </button>
            <button
              className="pill"
              onClick={[Function]}
            >
              bar
            </button>
            <button
              className="pill"
              onClick={[Function]}
            >
              baz
            </button>
            <button
              className="pill"
              onClick={[Function]}
            >
              qux
            </button>
          </div>,
          undefined,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="pill"
                onClick={[Function]}
              >
                foo
              </button>,
              <button
                className="pill"
                onClick={[Function]}
              >
                bar
              </button>,
              <button
                className="pill"
                onClick={[Function]}
              >
                baz
              </button>,
              <button
                className="pill"
                onClick={[Function]}
              >
                qux
              </button>,
            ],
            "className": "pillGroup",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "foo",
                "className": "pill",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "foo",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": "bar",
                "className": "pill",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "bar",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "host",
              "props": Object {
                "children": "baz",
                "className": "pill",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "baz",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "host",
              "props": Object {
                "children": "qux",
                "className": "pill",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "qux",
              "type": "button",
            },
          ],
          "type": "div",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Pill styles matches the snapshot 1`] = `
Object {
  "assistiveText": Object {
    "color": "#aaaaaa",
  },
  "bottom": Object {
    "height": 20,
    "marginTop": 4,
  },
  "container": Object {
    "marginBottom": 8,
    "position": "relative",
  },
  "disabled": Object {
    "& $label": Object {
      "color": "#dddddd",
    },
    "& $pill": Object {
      "background": "#ffffff",
      "borderColor": "#f5f5f5",
      "color": "#dddddd",
      "pointerEvents": "none",
    },
  },
  "errorAssistiveText": Object {
    "color": "#fe4a49",
  },
  "errorIcon": Object {
    "marginBottom": -2,
  },
  "label": Object {
    "& a": Object {
      "color": "#1476fb",
      "outline": "none",
      "textDecoration": "none",
    },
    "color": "#222222",
    "fontFamily": "'Open Sans', sans-serif",
    "fontSize": 12,
    "lineHeight": "20px",
    "transition": "0.3s all",
  },
  "left": Object {
    "float": "left",
  },
  "pill": Object {
    "&:first-child": Object {
      "borderRadius": "16px 0 0 16px",
    },
    "&:hover": Object {
      "background": "#fafafa",
    },
    "&:last-child": Object {
      "borderRadius": "0 16px 16px 0",
    },
    "&:not(:first-child)": Object {
      "marginLeft": -1,
    },
    "background": "#ffffff",
    "border": "1px solid #dddddd",
    "cursor": "pointer",
    "flex": 1,
    "fontFamily": "'Open Sans', sans-serif",
    "fontSize": 14,
    "height": 32,
    "lineHeight": "24px",
    "outline": 0,
    "padding": 0,
    "position": "relative",
    "transition": "0.3s all",
    "zIndex": 0,
  },
  "pillGroup": Object {
    "display": "flex",
  },
  "selected": Object {
    "background": Array [
      "rgba(8, 60, 174, 0.1)",
      "!important",
    ],
    "borderColor": "#083cae",
    "zIndex": 1,
  },
  "top": Object {
    "&:after": Object {
      "clear": "both",
      "content": "\\"\\"",
    },
    "height": 24,
  },
}
`;

exports[`PillJSS matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Pill)
    group={
      Array [
        Object {
          "id": 1,
          "label": "foo",
          "value": 10,
        },
        Object {
          "id": 2,
          "label": "bar",
          "value": 20,
        },
        Object {
          "id": 3,
          "label": "baz",
          "value": 30,
        },
        Object {
          "id": 4,
          "label": "qux",
          "value": 40,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "assistiveText": "Pill-assistiveText-0-1-10",
        "bottom": "Pill-bottom-0-1-9",
        "container": "Pill-container-0-1-1",
        "disabled": "Pill-disabled-0-1-6",
        "errorAssistiveText": "Pill-errorAssistiveText-0-1-11",
        "errorIcon": "Pill-errorIcon-0-1-12",
        "label": "Pill-label-0-1-3",
        "left": "Pill-left-0-1-4",
        "pill": "Pill-pill-0-1-7",
        "pillGroup": "Pill-pillGroup-0-1-5",
        "selected": "Pill-selected-0-1-8",
        "top": "Pill-top-0-1-2",
      },
      "group": Array [
        Object {
          "id": 1,
          "label": "foo",
          "value": 10,
        },
        Object {
          "id": 2,
          "label": "bar",
          "value": 20,
        },
        Object {
          "id": 3,
          "label": "baz",
          "value": 30,
        },
        Object {
          "id": 4,
          "label": "qux",
          "value": 40,
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "assistiveText": "Pill-assistiveText-0-1-10",
          "bottom": "Pill-bottom-0-1-9",
          "container": "Pill-container-0-1-1",
          "disabled": "Pill-disabled-0-1-6",
          "errorAssistiveText": "Pill-errorAssistiveText-0-1-11",
          "errorIcon": "Pill-errorIcon-0-1-12",
          "label": "Pill-label-0-1-3",
          "left": "Pill-left-0-1-4",
          "pill": "Pill-pill-0-1-7",
          "pillGroup": "Pill-pillGroup-0-1-5",
          "selected": "Pill-selected-0-1-8",
          "top": "Pill-top-0-1-2",
        },
        "group": Array [
          Object {
            "id": 1,
            "label": "foo",
            "value": 10,
          },
          Object {
            "id": 2,
            "label": "bar",
            "value": 20,
          },
          Object {
            "id": 3,
            "label": "baz",
            "value": 30,
          },
          Object {
            "id": 4,
            "label": "qux",
            "value": 40,
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
